scipy/spatial/transform/rotation.py

```diff
@@ -129,3 +129,56 @@ def as_dcm(self):
             return dcm[0]
         else:
             return dcm
+
+    @classmethod
+    def from_dcm(cls, mat, scalar=True):
+        # For now, mat must be a 3D matrix
+        num_rotations = mat.shape[0]
+        if scalar:
+            quat = np.empty((num_rotations, 4))
+            for rot_num in range(num_rotations):
+                A = mat[rot_num]
+                trA = A.trace()
+                choice = np.argmax(np.array([A[0, 0], A[1, 1], A[2, 2], trA]))
+
+                if choice in [0, 1, 2]:
+                    i = choice
+                    j = (i + 1) % 3
+                    k = (j + 1) % 3
+                    quat[rot_num, i] = 1 - trA + 2 * A[i, i]
+                    quat[rot_num, j] = A[j, i] + A[i, j]
+                    quat[rot_num, k] = A[k, i] + A[i, k]
+                    quat[rot_num, 3] = A[k, j] - A[j, k]
+                else:
+                    quat[rot_num] = np.array([
+                        A[2, 1] - A[1, 2],
+                        A[0, 2] - A[2, 0],
+                        A[1, 0] - A[0, 1],
+                        1 + trA])
+        else:
+            decision_matrix = np.zeros((num_rotations, 4))
```

Comment:
I think it's not the optimal way to implement that. I suggest the following approach.

1. You computed array of choices --- this is good.
2. Create the return array `quat = np.empty((n_rotations, 4))`.
3. Consider cases. 
```
ind = np.nonzero(choices != 3)[0]
i = choices[ind]
j = (i + 1) % 3
k = (j + 1) % 3
# Fill the quat[ind, 0], quat[ind, 1] in the same way as in scalar approach, use mat[ind, ...].

ind = np.nonzero(choices == 3)[0]
# Do the similar
```